src/prefect/core/task.py

```diff
@@ -173,7 +173,11 @@ def _reserved_attributes(self) -> Tuple[str]:
         computed to avoid circular import issues.
         """
         if not hasattr(Task, "_cached_reserved_attributes"):
-            Task._cached_reserved_attributes = tuple(sorted(Task().__dict__))  # type: ignore
+            # Create a base task instance to determine which attributes are reserved
+            # we need to disable the unused_task_tracker for this duration or it will
+            # track this task
+            with prefect.context(_unused_task_tracker=set()):
+                Task._cached_reserved_attributes = tuple(sorted(Task().__dict__))  # type: ignore
```

Comment:
We can also store the `Task()` object and `.remove()` it afterwards but this feels cleaner